.github/actions/macos-code-sign/action.yml

```diff
@@ -173,21 +169,95 @@ runs:
           submission_id=$(printf '%s\n' "$submission_json" | jq -r '.id // ""')
 
           if [[ -z "$submission_id" ]]; then
-            echo "Failed to retrieve submission ID for $binary"
+            echo "Failed to retrieve submission ID for $label"
             exit 1
           fi
 
-          echo "::notice title=Notarization::$binary submission ${submission_id} completed with status ${status}"
+          echo "::notice title=Notarization::$label submission ${submission_id} completed with status ${status}"
 
           if [[ "$status" != "Accepted" ]]; then
-            echo "Notarization failed for ${binary} (submission ${submission_id}, status ${status})"
+            echo "Notarization failed for ${label} (submission ${submission_id}, status ${status})"
+            exit 1
+          fi
+        }
+
+        notarize_binary() {
+          local binary="$1"
+          local source_path="codex-rs/target/${{ inputs.target }}/release/${binary}"
+          local archive_path="${RUNNER_TEMP}/${binary}.zip"
+
+          if [[ ! -f "$source_path" ]]; then
+            echo "Binary $source_path not found"
             exit 1
           fi
+
+          rm -f "$archive_path"
+          ditto -c -k --keepParent "$source_path" "$archive_path"
+
+          notarize_submission "$binary" "$archive_path"
+        }
+
+        build_dmg() {
+          local target="${{ inputs.target }}"
+          local release_dir="codex-rs/target/${target}/release"
+          local dmg_root="${RUNNER_TEMP}/codex-dmg-root"
+          local volname="Codex (${target})"
+          local dmg_path="${release_dir}/codex-${target}.dmg"
+
+          rm -rf "$dmg_root"
+          mkdir -p "$dmg_root"
+
+          for binary in codex codex-responses-api-proxy; do
+            local bin_path="${release_dir}/${binary}"
```

Comment:
I'm kind of struggling to follow the code, can you check that the files that we're putting inside the dmg are the ones that have already been signed and notarized?